### PR TITLE
Generate help pages from markdown

### PR DESCRIPTION
**Pull request by tzvipm**
## Summary
Generate pages for help book from markdown and JSON content. This will make it easier for newcomers to contribute help content to the project with confidence.

Part of #252 

## Checklist

- [x] Create generate script for help documentation
- [ ] Integrate generate script into build process
- [ ] Fix ordering of keyboard shortcuts
- [ ] Add myself to authors file

## Review Checklist
- [ ] I have responded to reviews and made changes where appropriate.
- [ ] I have tested the code
- [ ] I have updated comments / documentation related to the changes I made.
- [ ] I have rebased my PR branch onto xi-mac/master.

### CODE REVIEW

File: XiEditor.help/Contents/Resources/en.lproj/pages/KeyboardShortcuts.html [hunk 1]
```diff
@@ -1,40 +1,10 @@
...
-            <tr>
-                <td><kbd>&#8984;</kbd><kbd>F</kdb></td>
-                <td>Open find dialog</td>
-            </tr>
-        </table>
-    </body>
-</html>
+<h1>Keyboard Shortcuts</h1>
+
+<h2>Find and replace</h2>
```
Comment (tzvipm) [inline XiEditor.help/Contents/Resources/en.lproj/pages/KeyboardShortcuts.html]: I need to fix the ordering here

File: XiEditor.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -633,6 +634,26 @@
...
+			);
+			inputPaths = (
+			);
+			outputFileListPaths = (
+			);
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "cd $SRCROOT/XiEditor.help/Contents/Resources/en.lproj/gen\nswift -F . ./generate.swift\n";
```
Comment (tzvipm) [inline XiEditor.xcodeproj/project.pbxproj]: This doesn't work for some reason. I end up with a spinning help window 🤔

**Comment by tzvipm** [comment]
My idea, in terms of generating the help bundle, is to add 2 additional build targets:

1. `XiHelpGen`, which is a swift cli application that will be used to generate the help files
2. `XiHelp.help` which is a `.help` bundle that makes use of `XiHelpGen` to produce a `.help` bundle

and consume these in `XiEditor` (mac).

I managed to get (1) to build, but when I try to run it as part of the build process for (2), I get the following error:

```
Showing All Messages
PhaseScriptExecution Run\ Script /Users/hartleymelamed/Library/Developer/Xcode/DerivedData/XiEditor-egdbnnriprbpnqgjfirhowenfuts/Build/Intermediates.noindex/XiEditor.build/Debug/XiHelp.build/Script-31C3F7FB21D3D2300034AA2E.sh (in target: XiHelp)
    cd /Users/hartleymelamed/src/github.com/tzvipm/xi-mac

# ... export environment variables

dyld: Library not loaded: @rpath/Down.framework/Versions/A/Down
  Referenced from: /Users/hartleymelamed/Library/Developer/Xcode/DerivedData/XiEditor-egdbnnriprbpnqgjfirhowenfuts/Build/Products/Debug/XiHelpGen
  Reason: image not found
/Users/hartleymelamed/Library/Developer/Xcode/DerivedData/XiEditor-egdbnnriprbpnqgjfirhowenfuts/Build/Intermediates.noindex/XiEditor.build/Debug/XiHelp.build/Script-31C3F7FB21D3D2300034AA2E.sh: line 2: 35448 Abort trap: 6           $BUILT_PRODUCTS_DIR/XiHelpGen
Command PhaseScriptExecution failed with a nonzero exit code
```

If anyone has any thoughts on how to move forward with this, it would be greatly appreciated.

**Comment by jansol** [comment]
Sorry, won't have time to look into this in detail for at least another week. Anyhow it looks like a problem with how you are creating the XiHelpGen target. Is it a command line tool or a .app bundle? In the latter case it should have the framework copied into the bundle dir.

Isn't Down available via the Swift Package Manager that we are using? I'd expect getting it that way to do the right things. That failing you could of course bake it into the executable statically, either by building it as a static library or by simply dropping the sources into the XiHelpGen target.

**Comment by ryanbloom** [comment]
Not sure if anyone has seen this, but [jekyll-apple-help](https://github.com/chuckhoupt/jekyll-apple-help) looks like it could handle a lot of the markdown rendering and styling for us. (I haven't tried it, so I can't make a recommendation yet.)

**Comment by jansol** [comment]
Oh, that looks awesome! Wish I had known of it from the start. I think this also ties in nicely with our new website setup? Definitely am in favor of it in that case.


> the help developer documentation is byzantine and the tools are idiosyncratic

There are tools? 😆

**Comment by tzvipm** [comment]
Thanks @ryanbloom. I searched so hard for something like this, but apparently not hard enough 😛thanks for the resource!

**Comment by jansol** [comment]
@tzvipm Any news on this?

**Comment by tzvipm** [comment]
Thanks @jansol. I had some other commitments come up, but I would really like to contribute to Xi. I'll try to PR something this week.

**Comment by tzvipm** [comment]
Superseded by https://github.com/xi-editor/xi-mac/pull/440

**Pull request closed without merging**